jwst/wavecorr/wavecorr.py

```diff
@@ -73,26 +72,8 @@ def do_correction(input_model, wavecorr_file):
         # position information
         corrected = False
         if exp_type == 'NRS_FIXEDSLIT':
-            primary_slit = input_model.meta.instrument.fixed_slit
             for slit in output_model.slits:
                 if _is_point_source(slit, exp_type):
-                    # If fixed slit was not defined via MSA file,
-                    # it must have dither information to find the
-                    # source position, and it must be the primary slit
-                    if not _is_msa_fixed_slit(slit):
```

Comment:
I think the _is_msa_fixed_slit function is no longer used after this change and can be removed.

Reply:
Also, I think with this change the FS and MOS handling is now identical and we can remove the repeated code.